plugins/modules/ndo_tenant_igmp_snooping_policy.py

```diff
@@ -0,0 +1,359 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2024, Anvitha Jain (@anvjain) <anvjain@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+DOCUMENTATION = r"""
+---
+module: ndo_tenant_igmp_snooping_policy
+short_description: Manage Internet Group Management Protocol (IGMP) Snooping Policies in Tenant Policy Templates on Cisco Nexus Dashboard Orchestrator (NDO).
+description:
+- Manage IGMP Snooping Policies in Tenant Policy Templates on Cisco Nexus Dashboard Orchestrator (NDO).
+- This module is only supported on ND v3.1 (NDO v4.3) and later.
+author:
+- Anvitha Jain (@anvjain)
+options:
+  template:
+    description:
+    - The name of the tenant template.
+    type: str
+    aliases: [ tenant_template ]
+    required: true
+  name:
+    description:
+    - The name of the IGMP Snooping Policy.
+    type: str
+    aliases: [ igmp_snooping_policy ]
+  uuid:
+    description:
+    - The UUID of the IGMP Snooping Policy.
+    - This parameter is required when the IGMP Snooping Policy O(name) needs to be updated.
+    type: str
+  description:
+    description:
+    - The description of the IGMP Snooping Policy.
+    - Providing an empty string will remove the O(description="") from the IGMP Snooping Policy.
+    type: str
+  admin_state:
+    description:
+    - The administrative state of the IGMP Snooping Policy.
+    - Defaults to C(enabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  fast_leave_control:
+    description:
+    - The fast leave control of the IGMP Snooping Policy.
+    - Defaults to C(disabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  querier_control:
+    description:
+    - The querier control of the IGMP Snooping Policy.
+    - Defaults to C(disabled) when unset during creation.
+    type: str
+    choices: [ enabled, disabled ]
+  querier_version:
+    description:
+    - The querier version of the IGMP Snooping Policy.
+    - Defaults to C(v3) when unset during creation.
+    type: str
+    choices: [ v2, v3 ]
+  query_interval:
+    description:
+    - The query interval of the IGMP Snooping Policy.
```

Comment:
Should we also include the time period of all the interval options?
1. query_interval
2. query_response_interval
3. last_member_query_interval
4. start_query_interval

Reply:
Added time period